### PR TITLE
Added Platform support to PalWorld Chart

### PR DESCRIPTION
**Pull request by pbtcameron**
**Pull Request Summary:**

This pull request enhances the Helm configuration process for PalWorld by adding support for the new AllowConnectPlatform parameter. This parameter is required for configuring Xbox dedicated servers, a feature recently introduced in PalWorld.  The changes specifically focus on updating the Helm Questions to include prompts for configuring this new setting, ensuring seamless deployment of PalWorld on Kubernetes with Xbox server support.

**Key Changes:**

Helm Questions Update: The Helm Questions have been extended to incorporate the new AllowConnectPlatform parameter. This allows users to specify the desired platform connectivity options during Helm chart installation, enabling proper configuration of Xbox dedicated servers.

### CODE REVIEW

**Review by stavros-k — CHANGES_REQUESTED** [review]
Hello, please apply any changes only under the ix-dev/ directory.
Everything else is generated automatically.

Given that the migration is not written yet for this app, I can accept this change.
But this option should be also implemented in https://github.com/truenas/apps/blob/master/ix-dev/community/palworld/templates/docker-compose.yaml
In order to accept the PR here, otherwise the change can be done only on the truenas/apps repo.

Thanks

**Comment by pbtcameron** [comment]
> Hello, please apply any changes only under the ix-dev/ directory. Everything else is generated automatically.
> 
> Given that the migration is not written yet for this app, I can accept this change. But this option should be also implemented in https://github.com/truenas/apps/blob/master/ix-dev/community/palworld/templates/docker-compose.yaml In order to accept the PR here, otherwise the change can be done only on the truenas/apps repo.
> 
> Thanks

Hi,
Sorry, I was unaware. I'll make the changes there and make a PR.

Thanks

**Comment by pbtcameron** [comment]
> Hello, can you please remove changes in directories outside of `ix-dev` and sync changes we did on truenas/apps on this PR?
> 
> Thanks!

I've made the changes. Do I need to update the version to 1.0.5?

**Review by stavros-k — CHANGES_REQUESTED** [review]
Hello, I still see changes in directories outside of `ix-dev/`. Which shouldnt be happening.

Please revert those and only update files inside `ix-dev/`.

**Comment by pbtcameron** [comment]
> Hello, I still see changes in directories outside of `ix-dev/`. Which shouldnt be happening.
> 
> Please revert those and only update files inside `ix-dev/`.

Sorry, I'm a bit confused. By `ix-dev/` is it in reference to `Palworld/1.0.4/`, `Palworld/` or a whole other folder altogether?

**Comment by stavros-k** [comment]
Here: https://github.com/truenas/charts/tree/master/library/ix-dev/community/palworld

**Comment by pbtcameron** [comment]
> Here: https://github.com/truenas/charts/tree/master/library/ix-dev/community/palworld

Thanks. I've applied the changes

File: library/ix-dev/community/palworld/questions.yaml [hunk 1]
```diff
@@ -53,6 +53,18 @@ questions:
                   type: string
                   private: true
                   default: ""
+        - variable: allow_platform
```
Comment (stavros-k) [inline library/ix-dev/community/palworld/questions.yaml:56]:
```suggestion
        - variable: allowPlatform
```

File: library/ix-dev/community/palworld/questions.yaml [hunk 2]
```diff
@@ -53,6 +53,18 @@ questions:
...
+          label: Platform
+          description: |
+                  Specify which platforms are allowed to connect
```
Comment (stavros-k) [inline library/ix-dev/community/palworld/questions.yaml:59]:
```suggestion
          description: Specify which platforms are allowed to connect
```

File: library/ix-dev/community/palworld/templates/_palworld.tpl [hunk 3]
```diff
@@ -123,6 +132,7 @@ workload:
               set_ini_value "ServerDescription" {{ .Values.palworldConfig.server.description | quote }} true
               set_ini_value "ServerPassword" {{ .Values.palworldConfig.server.password | squote }} true false
               set_ini_value "AdminPassword" {{ .Values.palworldConfig.adminPassword | squote }} true false
+              set_ini_value "AllowConnectPlatform" {{ .Values.palworldConfig.allow_platform }} false
```
Comment (stavros-k) [inline library/ix-dev/community/palworld/templates/_palworld.tpl:135]:
```suggestion
              set_ini_value "AllowConnectPlatform" {{ .Values.palworldConfig.allowPlatform }} false
```

File: library/ix-dev/community/palworld/values.yaml [hunk 4]
```diff
@@ -14,6 +14,7 @@ palworldConfig:
     name: TrueNAS Palworld Server
     description: Palworld Server running on TrueNAS
     password: ''
+  allow_platform: Steam
```
Comment (stavros-k) [inline library/ix-dev/community/palworld/values.yaml:17]:
```suggestion
  allowPlatform: Steam
```

**Review by stavros-k — COMMENTED** [review]
Thanks! Small nit, lets keep the naming convention here consistent!
And we should be good to go!

**Review by stavros-k — APPROVED** [review]

**Comment by HDR-Performance** [comment]
Issue With Update for Palworld
attribute required

More info...
 Error: Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/middlewared/job.py", line 469, in run
    await self.future
  File "/usr/lib/python3/dist-packages/middlewared/job.py", line 511, in __run_body
    rv = await self.method(*args)
         ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3/dist-packages/middlewared/schema/processor.py", line 187, in nf
    return await func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3/dist-packages/middlewared/schema/processor.py", line 47, in nf
    res = await f(*args, **kwargs)
          ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3/dist-packages/middlewared/plugins/chart_releases_linux/upgrade.py", line 116, in upgrade
    await self.upgrade_chart_release(job, release, options)
  File "/usr/lib/python3/dist-packages/middlewared/plugins/chart_releases_linux/upgrade.py", line 267, in upgrade_chart_release
    config, context = await self.middleware.call(
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3/dist-packages/middlewared/main.py", line 1564, in call
    return await self._call(
           ^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3/dist-packages/middlewared/main.py", line 1417, in _call
    return await methodobj(*prepared_call.args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3/dist-packages/middlewared/plugins/chart_releases_linux/chart_release.py", line 353, in normalise_and_validate_values
    dict_obj = await self.middleware.call(
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3/dist-packages/middlewared/main.py", line 1564, in call
    return await self._call(
           ^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3/dist-packages/middlewared/main.py", line 1417, in _call
    return await methodobj(*prepared_call.args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3/dist-packages/middlewared/plugins/chart_releases_linux/validation.py", line 54, in validate_values
    verrors.check()
  File "/usr/lib/python3/dist-packages/middlewared/service_exception.py", line 70, in check
    raise self
middlewared.service_exception.ValidationErrors: [EINVAL] values.palworldConfig.allowPlatform: attribute required

**Comment by HDR-Performance** [comment]
how and where? in cnfg?

**Comment by stavros-k** [comment]
> how and where? in cnfg?

You dont have to do anything.
Wait for the linked PR to be merged!
Thanks

**Comment by HDR-Performance** [comment]
i cant update to the latest version is the issue nor is there an option for sever platform in 1.04 so it wont allow update to 1.05,

**Comment by stavros-k** [comment]
Just wait for the linked PR to be merged. Its not yet.
You will be able to upgrade AFTER that.

**Pull request merged**